tests/security/test_symlink_safety.py

```
@@ -422,48 +422,130 @@ def emit(self, record: logging.LogRecord) -> None:
+        finally:
+            stage.close()
+
+        assert result.failed, "stage must fail when category dir is a symlink"
+        assert not any(attacker_dir.iterdir()), "attacker dir must remain empty"
+        assert (
+            "security_event" in "".join(r.getMessage() for r in self._capture_log(stage))
+            or result.error is not None
+        ), "error must be set"
+
+    @staticmethod
+    def _capture_log(stage: object) -> list:
```

Ruff (0.15.13)

[warning] Unused static method argument: `stage`

(ARG004)